src/ray/object_manager/object_store_pool.cc

```diff
@@ -0,0 +1,40 @@
+#include "object_store_pool.h"
+
+namespace ray {
+
+ObjectStorePool::ObjectStorePool(std::string &store_socket_name) {
+  store_socket_name_ = store_socket_name;
+}
+
+std::shared_ptr<plasma::PlasmaClient> ObjectStorePool::GetObjectStore() {
+  pool_mutex.lock();
+  if (available_clients.empty()) {
+    Add();
+  }
+  std::shared_ptr<plasma::PlasmaClient> client = available_clients.back();
+  available_clients.pop_back();
+  pool_mutex.unlock();
+  return client;
+}
+
+void ObjectStorePool::ReleaseObjectStore(std::shared_ptr<plasma::PlasmaClient> client) {
+  pool_mutex.lock();
```

Comment:
actually, if readonly operations are common, we should try to use read/write locks instead.

Reply:
In both cases, `GetObjectStore` and `ReleaseObjectStore` both write to `available_clients`. Are there read/write locks for C++11?